File: app/src/main/java/com/renatojobal/libraryutpl/mainactivity/fsearchbook/ResultDao.java
package com.renatojobal.libraryutpl.mainactivity.fsearchbook;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Query;

import java.util.List;

/**
 * Dao for the result
 */
@Dao
public interface ResultDao {

    // Lets access to the result of search a book from the database
    // BookFull is a view table
    @Query("SELECT * FROM BookFull")
    LiveData<List<BookFull>> allResultLiveData();
}

File: app/src/main/java/com/renatojobal/libraryutpl/mainactivity/fsearchbook/ResultDao.kt
package com.renatojobal.libraryutpl.mainactivity.fsearchbook

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Query

@Dao
interface ResultDao {

    @Query(
            """
                SELECT * FROM SampleBookFull
            """
    )
    abstract fun allResultLiveData() : LiveData<List<SampleBookFull>>
}